### PR TITLE
libc/inttypes: Add posix api imaxdiv()

### PR DESCRIPTION
**Pull request by lokeshbv**
This patch adds support for libc inttypes imaxdiv().

Signed-off-by: Lokesh B V <lokesh.bv@partner.samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a7ca0b1760dbab0567646eea1137b898776d0dc5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a7ca0b1760dbab0567646eea1137b898776d0dc5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [90e1f5961cfbf33431790a0e57e306a6b7d18890] - Code Rule Check (C++) OK.

File: lib/libc/inttypes/lib_imaxdiv.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
```
Comment (jeongchanKim) [inline lib/libc/inttypes/lib_imaxdiv.c]: this is not needed

File: lib/libc/inttypes/lib_imaxdiv.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#include <inttypes.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+imaxdiv_t imaxdiv(intmax_t num, intmax_t den)
```
Comment (jeongchanKim) [inline lib/libc/inttypes/lib_imaxdiv.c]: where is the prototype for this api?

**Comment by seinfra** [comment]
Target : [40ebcc5dd6ed215a8df0c1873f9636ae5ef9c894] -

**Comment by seinfra** [comment]
Target : [40ebcc5dd6ed215a8df0c1873f9636ae5ef9c894] -

File: lib/libc/inttypes/lib_imaxdiv.c (hunk 2 above)
Comment (lokeshbv) [reply lib/libc/inttypes/lib_imaxdiv.c]: https://github.com/Samsung/TizenRT/blob/90713d931de9057641dc915a38db6967e7fa5e38/os/include/inttypes.h#L214

**Comment by seinfra** [comment]
Target : [3417b3ce0061b5725be473c407b9ae117fd86d4b] -

**Comment by seinfra** [comment]
Target : [3417b3ce0061b5725be473c407b9ae117fd86d4b] -

**Comment by seinfra** [comment]
Target : [066c5db6f2b6071da2d96344e9013c002274a0f7] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [066c5db6f2b6071da2d96344e9013c002274a0f7] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_inttypes.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+ ****************************************************************************/
+
+/// @file tc_libc_inttypes.c
+
+/// @brief Test Case Example for Libc Stdlib API
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
```
Comment (yashwanth686007) [inline apps/examples/testcase/le_tc/kernel/tc_libc_inttypes.c]: Is this header required? Since there are no configs in the file

Comment (lokeshbv) [reply apps/examples/testcase/le_tc/kernel/tc_libc_inttypes.c]: No, its's not required and `errno.h` as well. I will remove both

**Comment by sunghan-chang** [comment]
@lokeshbv Do you get any requirement for this? Is it needed usaually? As you know, we need to optimize RAM and ROM usages, but there is no way to disable inttypes APIs. We had better support a way among below options.
1. Support config to disable it.
2. Not add new api whiich is not needed usaually
3. else any good suggestion

**Comment by seinfra** [comment]
Target : [a8f57c16c4b0e4fa9478e28a4251a8ca34e40120] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a8f57c16c4b0e4fa9478e28a4251a8ca34e40120] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_mqueue.c [hunk 4]
```diff
@@ -16,41 +16,9 @@
  *
  ****************************************************************************/
 
-/// @file mqueue.c
+/// @file tc_mqueue.c
+
 /// @brief Test Case Example for Message Queue API
-/**************************************************************************
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_mqueue.c]: This tc was come from mqueue.c at kernel_sample and kernel_sample was renamed from ostest of Nuttx. This is almost same as mqueue of kernel_sample. We should not remove Nuttx license.

**Comment by sunghan-chang** [comment]
@lokeshbv As we discussed, please ignore my previous comment which says ROM & RAM usages.

File: apps/examples/testcase/le_tc/kernel/tc_mqueue.c (hunk 4 above)
Comment (lokeshbv) [reply apps/examples/testcase/le_tc/kernel/tc_mqueue.c]: yes, Nuttx license has to be there I will revert it

**Comment by seinfra** [comment]
Target : [2b6564020f7b057d8dda5554a037894437069bcb] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2b6564020f7b057d8dda5554a037894437069bcb] - Code Rule Check (C++) OK.

**Pull request merged**